test/expressions/comparisons/paramUintCompares.chpl

```diff
@@ -0,0 +1,45 @@
+//
+// These uint comparisons should all be compile-time folded false
+//
+proc testit(type t) {
+  var x: t = 2;
+  param zero: t = 0;
+
+  if (x < 0) then
+    compilerError("This should always be false");
+
+  if (!(x >= 0)) then
+    compilerError("This should always be false too");
+
+  if (0 > x) then
+    compilerError("As should this");
+
+  if (!(0 <= x)) then
+    compilerError("And also this");
+
+  if (x < zero) then
+    compilerError("This should always be false");
+
+  if (!(x >= zero)) then
+    compilerError("This should always be false too");
+
+  if (zero > x) then
+    compilerError("As should this");
+
+  if (!(zero <= x)) then
+    compilerError("And also this");
+}
+
+//
+// Test comparisons for all uints
+//
+testit(uint);
+testit(uint(64));
+testit(uint(32));
+testit(uint(16));
+testit(uint(8));
+
+//
+// This isn't actually an error, but avoids paying any execution time
+//
+compilerError("Everything's OK!");
```

Comment:
Seems weird to try to avoid execution time cost, when it'll be far, far faster than compilation times. If you want to skip execution, maybe use a .noexec instead?